ipfs-gateway-finder/src/main.rs

```diff
@@ -199,6 +202,18 @@ async fn do_probing() -> Result<()> {
             )))
         }
     };
+
+    gateway_list = gateway_list
+        .into_iter()
```

Comment:
You can merge this with the part above by just appending everything from `.map` to the closing curly brace of the `match` expression. That avoids the mutable binding, which is always nice.
Alternatively, if that doesn't look nice, something like this also works:

```rust
let a: Vec<String> = {
    let mut a_mut = foo();
    a_mut.into_iter().do_stuff().collect()
}
```
This keeps the mutability inside the block.